app/scripts/modules/core/src/presentation/hooks/useDimensions.hook.ts

```diff
@@ -0,0 +1,57 @@
+import { debounce } from 'lodash';
+import React from 'react';
+
+export const useWindowDimensions = ({ delay = 200, isActive = true }: { delay?: number; isActive?: boolean }) => {
+  const [dimension, setDimension] = React.useState({ width: window.innerWidth, height: window.innerHeight });
+
+  React.useLayoutEffect(() => {
+    const debouncedResizeHandler = debounce(() => {
+      setDimension({ width: window.innerWidth, height: window.innerHeight });
+    }, delay);
+    if (isActive) {
+      window.addEventListener('resize', debouncedResizeHandler);
+      return () => window.removeEventListener('resize', debouncedResizeHandler);
+    } else {
+      return () => {};
+    }
+  }, [delay, isActive]);
+
+  React.useLayoutEffect(() => {
+    if (isActive) {
+      setDimension({ width: window.innerWidth, height: window.innerHeight });
+    }
+  }, [isActive]);
```

Comment:
It would be helpful to add comments describing what is going on here. Why are there two uses of `useLayoutEffect` and what does each one of them do? Also describe the purpose of `delay` and `isActive`. I can't quite figure out what they do within the context of this file.

Reply:
Just confirmed offline that this is NOT used at all. So going to ignore it and look at the rest